k2/csrc/fsa_algo.h

```diff
@@ -144,9 +146,50 @@ bool Connect(const Fsa &a, Fsa *b, std::vector<int32_t> *arc_map = nullptr);
                     figure out the details).  Note: the output FSA's weights can be
                     computed later on, by calling code, using the info in arc_map.
  */
-void RmEpsilonsPruned(const WfsaWithFbWeights &a, float beam, Fsa *b,
+void RmEpsilonsPrunedMax(const WfsaWithFbWeights &a, float beam, Fsa *b,
+                         std::vector<std::vector<int32_t>> *arc_map);
+
+/*
+  Version of RmEpsilonsPrunedMax that doesn't support pruning; see its
+  documentation.
+ */
+void RmEpsilonsMax(const Fsa &a, float *a_weights, Fsa *b,
```

Comment:
The interfaces if OK for now. I guess we may need to define `semiring operation` (max/log-sum) to avoid duplicate code when we implement these alogrithms, what do you think?

Reply:
RE semirings: it's not quite as simple as that, as the algorithms needed to compute the derivative info can't, I think, be super easily templated.  The structures they require are a bit different.  As I continue implementing determinization I'll see if there are commonalities though.

Reply:
Got it, thanks!